### ISSUE TITLE
Help needed for using ccall to implment a CUDA kernel

### ISSUE THREAD

**Issue by xukai92** [JuliaGPU/CUDAnative.jl#243]
I'm looking into implementing a kernel for the the digamma function according to this (https://github.com/JuliaMath/SpecialFunctions.jl/blob/master/src/gamma.jl#L51). I don't know the correct way to translate the ccall.

I attempted to do below
```julia
function kernel_digamma(a, out)
  i = (blockIdx().x - 1) * blockDim().x + threadIdx().x
  x = BigFloat(a[i])
  z = BigFloat()
  ccall((:mpfr_digamma, :libmpfr), Int32, (Ref{BigFloat}, Ref{BigFloat}, Int32), z, x, ROUNDING_MODE[])
  out[i] = Float32(z)
end

function digamma(a)
  out = similar(a)
  @cuda kernel_digamma(a, out)
  return out
end
```

But it doesn't seem to work. Can anyone give me some help please?

**Comment by xukai92** [comment]
Yes I guess that's what I should use instead. Do you know which library I should looking for?

**Comment by jekbradbury** [comment]
You want to use or port the method above with signature `function digamma(z::ComplexOrReal{Float64})`

**Comment by xukai92** [comment]
Does it mean I should use another C libirary which implements the float and double version of digamma?
